src/core/instance.core.ts

```diff
@@ -172,40 +172,54 @@ export class ZoomPanPinch {
     wrapper.addEventListener("touchend", this.onTouchPanningStop, passive);
   };
 
-  handleInitialize = (contentComponent: HTMLDivElement): void => {
+  handleInitialize = (wrapper: HTMLDivElement, contentComponent: HTMLDivElement): void => {
+    let frameId: number;
     let isCentered = false;
     
     const { centerOnInit } = this.setup;
 
+    const hasTarget = (entries: ResizeObserverEntry[], target: Element) => {
+      for(const entry of entries) {
+        if(entry.target === target) {
+          return true;
+        }
+      }
+
+      return false;
+    }
+
     this.applyTransformation();
     this.onInitCallbacks.forEach((callback) => {
       callback(getContext(this));
     });
 
     this.observer = new ResizeObserver((entries) => {
-      for(const entry of entries) {
-        if(entry.target === contentComponent) {
-          if(centerOnInit && !isCentered) {
-            const currentWidth = contentComponent.offsetWidth;
-            const currentHeight = contentComponent.offsetHeight;
-    
-            if (currentWidth > 0 || currentHeight > 0) {
-              isCentered = true;
-    
-              this.setCenter();
-            }
-          } else {
-            const { pinchMidpoint } = this;
-    
-            handleAlignToScaleBounds(this, pinchMidpoint?.x, pinchMidpoint?.y);
+      if(hasTarget(entries, wrapper) || hasTarget(contentComponent)) {
+        if(centerOnInit && !isCentered) {
+          const currentWidth = contentComponent.offsetWidth;
+          const currentHeight = contentComponent.offsetHeight;
+  
+          if (currentWidth > 0 || currentHeight > 0) {
+            isCentered = true;
+  
+            this.setCenter();
           }
+        } else {
+          cancelAnimationFrame(frameId);
 
-          break;
-        }
+          frameId = requestAnimationFrame(() => {
+            const { scale } = this.transformState;
+
+            handleCancelAnimation(this);
+            handleCalculateBounds(this, scale);
+            handleAlignToBounds(this);
```

Comment:
I would add the `customTime` parameter to handleAlignToBounds so it would be:

```ts
handleAlignToBounds(this, 0)
```

So the alignment will be instant - or controlled via new prop `resizeAlignmentTime: number`

Reply:
This is a very good implementation👍.

Reply:
Can you add it here too?

Reply:
There is no need for it to limit the mouse panning. Check this out:

```ts
export function handleAlignToBounds(
  contextInstance: ReactZoomPanPinchContext,
  // one addition here! it is still optional
  customAlignmentTime?: number
): void {
  const { scale } = contextInstance.state;
  const { minScale, autoAlignment } = contextInstance.setup;
  const { disabled, sizeX, sizeY, animationTime, animationType } =
    autoAlignment;

  const isDisabled = disabled || scale < minScale || (!sizeX && !sizeY);

  if (isDisabled) return;

  const targetState = handlePanToBounds(contextInstance);

  if (targetState) {
    animate(
       contextInstance, 
       targetState, 
       // and here! we only apply it for cases where customAlignment is needed
       customAlignmentTime ?? animationTime, 
       animationType
    );
  }
}
```

This way there is no need for the setTimeout 😄

Reply:
Btw. Thank you for your work, that's a great contribution! 🙌🏻